src/main/java/com/dawkastresu/medicalclinic/PatientService.java

```diff
@@ -0,0 +1,66 @@
+package com.dawkastresu.medicalclinic;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+
+//ReauiredArgsConstruktor generuje konstruktor który przyjmuje pola finalne
+@RequiredArgsConstructor
+//Adnotacja Service mówi że ta klasa jest beanem zawierajacym logike biznesową aplikacji
+@Service
+public class PatientService {
+    //PatientRepository również musi być beanem aby Spring mógł dostarczyć instancję tej klasy
+    private final PatientRepository patientRepository;
+
+    public PatientDto editByEmail(String email, CreatePatientCommand createPatientCommand) {
+        Patient newPatient = PatientMapper.mapToPatient(createPatientCommand);
+        PatientValidator.newValueNotNullValidate(newPatient);
+        PatientValidator.validatePatientEdit(newPatient);
+        Patient patient = patientRepository.findByEmail(email)
+                .orElseThrow(() -> new PatientNotFoundException("Patient not found", HttpStatus.NOT_FOUND));
+
+        PatientValidator.cardIdNrNotChangedValidate(patient, newPatient);
+
+        patient.setEmail(newPatient.getEmail());
```

Comment:
Przenieś ustawianie nowych wartości do klasy Patient. Utwórz tam metodę update, która będzie przyjmować jako parametr Patient -> 

void update(Patient patient) {
this.setEmail(patient.getEmail())
this.setFirstName(patient.getFirstName()) <- i tak dalej do wszystkich pól
...
}